c7n/resources/cloudfront.py

```diff
@@ -181,9 +182,130 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('check-origin')
+class CheckOrigin(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront, and verify ownership.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: cloudfront-origin-check
+                resource: distribution
+                filters:
+                  origins:
+                    - S3
+                    - CUSTOM
+                    - S3-CUSTOM
+                  accounts:
+                    - c7n-test-domain.com
+                  accounts_from:
+                    url: *accounts-list
+                    expr: accounts-list.canonical_id
+    """
+
+    schema = type_schema(
+        'check-origin',
+        origins={'type': 'array', 'items': {
+            'type': 'string',
+            'enum': ['S3', 'CUSTOM', 'S3-CUSTOM']}
+        },
+        accounts={'type': 'array', 'items': {'type': 'string'}},
+        accounts_from=ValuesFrom.schema)
+
+    permissions = ('s3:GetBucketAcl',)
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def process_s3_target(self, r, target_bucket, client, accounts):
+        try:
+            b = client.get_bucket_acl(
+                Bucket=target_bucket
+            )
+            self.log.debug("Target bucket %s exists." % target_bucket)
+            if accounts and b['Owner']['ID'] not in accounts:
+                self.log.debug("Bucket %s owner not in accounts whitelist." % target_bucket)
```

Comment:
why is this just logging that the account isn't in the whitelist? if we're augmenting the resource for when it is in the whitelist we should also be doing so when the resource isn't.